yasa/evaluation.py

```diff
@@ -115,123 +109,403 @@ class EpochByEpochEvaluation:
+    N3        34   26   62  42   15    0    0    179
+    REM       15   12   13  10    0    0    0     50
+    ART        0    0    0   0    0    0    0      0
+    UNS        0    0    0   0    0    0    0      0
+    Total    277  128  333  98  124    0    0    960
     """
-    def __init__(self, hypno_ref, hypno_test):
-        from yasa.hypno import Hypnogram  # Loading here to avoid circular import
-        assert isinstance(hypno_ref, Hypnogram), "`hypno_ref` must be a YASA Hypnogram"
-        assert isinstance(hypno_test, Hypnogram), "`hypno_test` must be a YASA Hypnogram"
-        assert hypno_ref.n_stages == hypno_test.n_stages, (
-            "`hypno_ref` and `hypno_test` must have the same `n_stages`")
-        if (n_ref := hypno_ref.n_epochs) != (n_test := hypno_test.n_epochs):
-            ## NOTE: would be nice to have a Hypnogram.trim() method for moments like this.
-            if n_ref > n_test:
-                hypno_ref = Hypnogram(hypno_ref.hypno[:n_test], n_stages=hypno_ref.n_stages)
-                n_trimmed = n_ref - n_test
-                warn_msg = f"`hypno_ref` longer than `hypno_test`, trimmed to {n_test} epochs"
-            else:
-                hypno_test = Hypnogram(hypno_test.hypno[:n_ref], n_stages=hypno_test.n_stages)
-                n_trimmed = n_test - n_ref
-                warn_msg = f"`hypno_test` longer than `hypno_ref`, {n_trimmed} epochs trimmed"
-            ## Q: Should be downplayed as INFO?
-            logger.warning(warn_msg)
-        self.hypno_ref = hypno_ref
-        self.hypno_test = hypno_test
-
-    def get_agreement(self):
+
+    def __init__(self, refr_hyps, test_hyps):
+        from yasa.hypno import Hypnogram  # Avoiding circular import
+
+        assert hasattr(refr_hyps, "__iter__"), "`refr_hyps` must be a an iterable"
+        assert hasattr(test_hyps, "__iter__"), "`test_hyps` must be a an iterable"
+        assert type(refr_hyps) == type(test_hyps), "`refr_hyps` and `test_hyps` must be same type"
+        assert len(refr_hyps) == len(
+            test_hyps
+        ), "`refr_hyps` and `test_hyps` must have the same number of hypnograms"
+
+        if isinstance(refr_hyps, dict):
+            # If user provides dictionaries, split into sleep IDs and hypnograms
+            assert (
+                refr_hyps.keys() == test_hyps.keys()
+            ), "hypnograms in `refr_hyps` and `test_hyps` must have identical sleep IDs"
+            sleep_ids, refr_hyps = zip(*refr_hyps.items())
+            test_hyps = tuple(test_hyps.values())
+        else:
+            # Create hypnogram_ids
+            sleep_ids = tuple(range(1, 1 + len(refr_hyps)))
+
+        assert all(
+            isinstance(hyp, Hypnogram) for hyp in refr_hyps + test_hyps
+        ), "`refr_hyps` and `test_hyps` must only include YASA hypnograms"
+        assert all(
+            h.scorer is not None for h in refr_hyps + test_hyps
+        ), "all hypnograms must have a scorer name"
```

Comment:
Could this just default to "sleep_id" or "sleep_ids_refr/test" instead?

Reply:
Just to clarify, do you mean instead of `refr_hyps` and `test_hyps`? And you think this should also be applied to the attribute names, such that they are accessible as `ebe.refr` and `ebe.test`?

Reply:
Sorry about the confusion  — I meant that we are currently forcing the users to include a `scorer` name when creating their yasa.Hypnogram instance. However, we could have default values for `scorer`, which would be "test" and "ref" based on which sequence of Hypnogram the current instance belongs to.

Reply:
Right, I initially had those as arguments. (`test_name` and `reference_name` with defaults as "Test" and "Reference"). It's an easy switch back. I leaned towards the current because I thought it helped show the user about the `scorer` attribute and what it should be used for. I mean, force to submit to the power of the Hypnogram?

But ya what you prefer and I'll adjust :)

Reply:
Let's leave it like that for now! I need to try this out when I do the full PR review :)